watson_developer_cloud/websocket/speech_to_text_websocket_listener.py

```diff
@@ -0,0 +1,174 @@
+            super(self.__class__, self).__init__()
+
+        def build_start_message(self, options):
+            options['action'] = 'start'
+            return options
+
+        def build_close_message(self):
+            return json.dumps({'action': 'close'})
+
+        # helper method that sends a chunk of audio if needed (as required what the specified pacing is)
+        def send_audio(self, data):
+            def send_chunk(chunk, final=False):
+                self.bytes_sent += len(chunk)
+                self.sendMessage(chunk, isBinary=True)
+                if final:
+                    self.sendMessage(b'', isBinary=True)
+
+            if (self.bytes_sent + self.ONE_KB >= len(data)):
+                if (len(data) > self.bytes_sent):
+                    send_chunk(data[self.bytes_sent:len(data)], True)
+                    return
+
+            send_chunk(data[self.bytes_sent:self.bytes_sent + self.ONE_KB])
+            self.factory.reactor.callLater(0.01, self.send_audio, data=data)
+            return
+
+        def extract_transcripts(self, alternatives):
+            transcripts = []
+            for alternative in alternatives:
+                transcript = {}
+                if 'confidence' in alternative:
+                    transcript['confidence'] = alternative['confidence']
+                transcript['transcript'] = alternative['transcript']
+                transcripts.append(transcript)
+            return transcripts
+
+        def onConnect(self, response):
+            log.msg("onConnect, server connected: {}".format(response.peer))
+            self.callback.on_connected()
+
+        def onOpen(self):
+            log.msg("onOpen, Websocket connection open")
+            # send the initialization parameters
+            init_data = self.build_start_message(self.options)
+            self.sendMessage(json.dumps(init_data).encode('utf8'))
+
+            # start sending audio right away (it will get buffered in the STT service)
+            self.send_audio(self.audio.read())
+            log.msg("onOpen ends")
+
+        def onMessage(self, payload, isBinary):
+            json_object = json.loads(payload.decode('utf8'))
+
+            if 'error' in json_object:
+                log.msg('Error : {}'.format(json_object['error']))
+                # Only call on_error() if a real error occurred. The STT service sends
+                # // {"error" : "No speech detected for 5s"} for valid timeouts, configured by
+                # options.inactivity_timeout
+                error = json_object['error']
+                if error.startswith(self.TIMEOUT_PREFIX):
+                    self.callback.on_inactivity_timeout()
+                else:
+                    self.callback.on_error()
+
+            # if uninitialized, receive the initialization response from the server
+            elif 'state' in json_object:
+                self.listening_messages += 1
```

Comment:
You will only get up to 1 listening message so I think you should use a boolean. What you want to know is if it's the first message which means it's listening or if it's the second message which says it's done processing the audio file.